### ISSUE TITLE
Runner time limit

### ISSUE THREAD

**Issue by 7adamf** [runs-on/runs-on#195]
Is there a way to set a time limit on runners - for example if a job gets stuck the cost can be high - would be nice to be able to terminate a runner after x hours.

**Comment by crohr** [comment]
@7adamf the default global timeout is 12h (the instance will auto-shutdown after that time), but currently there is no way to override it. For your use-case, would you like an override at the stack level (i.e. for all runners), or at the runner level, with an additional `runs-on:` label?

**Comment by 7adamf** [comment]
Yes that would work great if possible!

**Comment by colstrom** [comment]
I think this is a case where stack-level and label-level overrides are both valuable, but I suspect they'd be used a bit differently.

- at the stack level feels like a safety measure, and 12h seems like a very reasonable default. I imagine this would be overridden to **increase** the limit, for orgs that have particularly long running things like soak tests.
- at the label level, on the other hand, I'd expect this to be used more to **reduce** the limits, for jobs with expected short durations. For instance, if rendering some markdown files to static HTML takes more than an hour, something has probably gone horribly wrong (or you have a terrifying amount of markdown).

🤔

**Comment by crohr** [comment]
Makes sense. Just trying to find a way to not duplicate existing attributes in workflows / add too many labels. GitHub already has timeout-minutes so ideally RunsOn would be able to infer the job timeout from that attribute if specified.

**Comment by crohr** [comment]
The original issue was:

> Is there a way to set a time limit on runners - for example if a job gets stuck the cost can be high - would be nice to be able to terminate a runner after x hours.

I believe it's just better to use the `timeout-minutes` from GitHub. I've never seen it NOT terminate the runner if it's correctly set yet. So adding another label for RunsOn would not be a meaningful improvement here.

Going to close for now, but let me know if I'm wrong here!
